src/traces/bar/cross_trace_calc.js

```diff
@@ -201,12 +214,12 @@ function setGroupPositionsInGroupMode(gd, pa, sa, calcTraces) {
 function setGroupPositionsInStackOrRelativeMode(gd, pa, sa, calcTraces) {
     var fullLayout = gd._fullLayout;
     var barmode = fullLayout.barmode;
-    var stack = barmode === 'stack';
-    var relative = barmode === 'relative';
     var barnorm = fullLayout.barnorm;
-    var separateNegativeValues = relative;
-    var dontMergeOverlappingData = !(barnorm || stack || relative);
-    var sieve = new Sieve(calcTraces, separateNegativeValues, dontMergeOverlappingData);
+
+    var sieve = new Sieve(calcTraces, {
+        separateNegativeValues: barmode === 'relative',
+        dontMergeOverlappingData: !(barnorm || barmode === 'stack' || barmode === 'relative')
```

Comment:
The function now has a more clear interface block which is great. 
Now that the keys would be presented in the `.min` files could we zip `separateNegativeValues` into e.g. `separateNegVals` and `dontMergeOverlappingData` into a short form?

Reply:
Done in c019c5ca8